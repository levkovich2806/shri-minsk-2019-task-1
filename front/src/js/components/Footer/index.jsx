import React, { PureComponent } from "react";
import styles from "./index.module.scss";

class Footer extends PureComponent {
  render() {
    const { author, copyright } = this.props;
    return (
      <div className={styles.footer}>
        <div className={styles.footer__author}>{author}</div>
        <div className={styles.footer__copyright}>&copy; {copyright}</div>
      </div>
    );
  }
}

export default Footer;
