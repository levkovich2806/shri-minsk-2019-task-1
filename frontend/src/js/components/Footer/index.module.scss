.footer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px;
  background: #fff;
  color: rgba(51, 51, 51, 0.6);
  font-family: Helvetica, Arial, sans-serif;
  font-size: 16px;
  line-height: 32px;
}

@media (min-width: 1100px) {
  .footer {
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    padding: 16px;
    background: #fff;
    color: rgba(51, 51, 51, 0.6);
    font-family: Helvetica, Arial, sans-serif;
    font-size: 16px;
    line-height: 32px;

    &__author {
      margin: 0 auto;
      padding-left: 120px;
    }
  }
}
