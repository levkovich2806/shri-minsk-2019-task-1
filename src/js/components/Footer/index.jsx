import React, { PureComponent } from "react";
import styles from "./index.module.scss";

class Footer extends PureComponent {
  render() {
    return (
      <div className={styles.footer}>
        <div className={styles.footer__author}>Сергей Левкович</div>
        <div className={styles.footer__copyright}>&copy; Яндекс ШРИ</div>
      </div>
    );
  }
}

export default Footer;
