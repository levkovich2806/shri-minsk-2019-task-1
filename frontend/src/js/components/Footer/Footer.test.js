import React from "react";
import { mount } from "enzyme";
import Footer from "./index.jsx";
import renderer from "react-test-renderer";

describe("Footer", function() {
  test("Footer rendering", () => {
    const params = {
      author: "Сергей Левкович",
      copyright: "Яндекс ШРИ"
    };
    const component = renderer.create(<Footer {...params} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Footer renders the text inside it", () => {
    const params = {
      author: "Сергей Левкович",
      copyright: "Яндекс ШРИ"
    };
    const wrapper = mount(<Footer {...params} />);
    const p = wrapper.find(".footer__author");
    expect(p.text()).toBe("Сергей Левкович");
  });
});
