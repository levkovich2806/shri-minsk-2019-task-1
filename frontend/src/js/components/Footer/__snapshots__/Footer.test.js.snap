// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer rendering 1`] = `
<div
  className="footer"
>
  <div
    className="footer__author"
  >
    Сергей Левкович
  </div>
  <div
    className="footer__copyright"
  >
    © 
    Яндекс ШРИ
  </div>
</div>
`;
